test/topology/conftest.py

```diff
@@ -226,3 +264,24 @@ def skip_mode_fixture(request, mode):
     for reason, platform_key in skipped_funcs.get((request.function, mode), []):
         if platform_key is None or platform_key in platform.platform():
             pytest.skip(f'{request.node.name} skipped, reason: {reason}')
+
+
+def pytest_configure(config):
+    config.pluginmanager.register(ReportPlugin())
+
+
+
+class ReportPlugin:
+    # Pytest hook to modify test name to include mode and run_id
+    def pytest_configure(self, config):
+        self.config = config
+        self.mode = config.getoption("mode")
+        self.run_id = config.getoption("run_id")
+
+    @pytest.hookimpl(hookwrapper=True)
+    def pytest_runtest_makereport(self):
+        outcome = yield
+        report = outcome.get_result()
+        for param in [self.mode, self.run_id]:
+            if param:
```

Comment:
Is it possible we don't have `mode` or `run_id`?

Reply:
mode is required
```
 parser.addoption('--mode', action='store', required=True,
                     help='Scylla build mode. Tests can use it to adjust their behavior.')
```
run_id is None by default
```
  parser.addoption('--run_id', action='store', default=None,
```
so "if param" will false in the case of Node

Reply:
Agreed with using default -- and then we can remove this weird `for` loop, instead do `report.node_id = f"{report.nodeid}.{self.mode}.{self.run_id or 1}"` or sth

(but probably better to assign `1` in constructor when assigning `self.run_id` and then it doesn't have to be `Optional`)